client/src/components/UpLoadImages/index.tsx

```diff
@@ -31,57 +49,105 @@ function UploadFiles() {
     setFile(fileList);
   };
   const handleUpload = () => {
-    if (!file) {
-      setSnackData({ type: 'info', message: 'You should choose an image befor upload' });
-      setOpen(true);
-    } else {
+    if (file) {
       setLoading(true);
       const toUploadfiles = [];
       for (let i = 0; i < file.length; i += 1) {
-        const storageRef = ref(storage, `/Images/cars/carId/${file[i].name}`);
-        const touploadFile = uploadBytes(storageRef, file[i]);
-        toUploadfiles.push(touploadFile);
+        const storageRef = ref(storage, `/Cars' Images/${carId}/${file[i].name}`);
+        const toUploadFile = uploadBytes(storageRef, file[i]);
+        toUploadfiles.push(toUploadFile);
       }
       Promise.all(toUploadfiles)
```

Comment:
```suggestion
    try {
        const snapshots = await Promise.all(toUploadfiles);
        const toDownloadUrls = await snapshots.map(({ ref: snapshotRef }) => getDownloadURL(snapshotRef));
        const urls = await Promise.all(toDownloadUrls);
        urls.map((url) => saveImage(url));
        const result = await httpInstance.put(`/cars/${carId}`, { state: 'on-market' });
        if (fileInput.current) {
          fileInput.current.value = null;
        }
        setFile(undefined);
        setLoading(false);
        setSnackData({ type: 'success', message: 'Uploaded successfully' });
        setOpen(true);
        navigate('/admin');
      } catch (error) {
        setSnackData({ type: 'error', message: 'Somthing went wrong' });
        setOpen(true);
      }
```
in our project we use async and await you here use then because that there is nested then I suggestion to use async and await it is easier than then and more tidy